src/FulfillerMultiSelect.jsx

```diff
@@ -100,17 +115,31 @@ class FulfillerMultiSelect extends React.Component {
     }
 
     handleChange(e) {
-        const fulfillerIds = e && e.map(f => f.fulfiller ? f.fulfiller.fulfillerId : f.fulfillerId);
-        if (!e || fulfillerIds === this.state.selectedFulfillerIds) {
+        if (!e) {
+            return;
+        }
+
+        const fulfillerIds = this.props.multi
+            ? e.map(f => f.fulfiller
+                ? f.fulfiller.fulfillerId
+                : f.fulfillerId
+            )
+            : [e.fulfiller.fulfillerId];
+
+        if (fulfillerIds === this.state.selectedFulfillerIds) {
             return;
```

Comment:
Apparently [] === [] returns `false`. Got to do an object-by-object comparison here.